domain/application/state/unit.go

```diff
@@ -1423,3 +1426,321 @@ ON CONFLICT(unit_uuid) DO UPDATE SET
 	}
 	return nil
 }
+
+// GetModelConstraints returns the currently set constraints for the model.
+// The following error types can be expected:
+// - [modelerrors.NotFound]: when no model exists to set constraints for.
+// - [modelerrors.ConstraintsNotFound]: when no model constraints have been
+// set for the model.
+// Note: This method should mirror the model domain method of the same name.
+func (st *State) GetModelConstraints(
+	ctx context.Context,
+) (constraints.Constraints, error) {
+	db, err := st.DB()
+	if err != nil {
+		return constraints.Constraints{}, errors.Capture(err)
+	}
+
+	selectTagStmt, err := st.Prepare(
+		"SELECT &dbConstraintTag.* FROM v_model_constraint_tag", dbConstraintTag{},
+	)
+	if err != nil {
+		return constraints.Constraints{}, errors.Capture(err)
+	}
+
+	selectSpaceStmt, err := st.Prepare(
+		"SELECT &dbConstraintSpace.* FROM v_model_constraint_space", dbConstraintSpace{},
+	)
+	if err != nil {
+		return constraints.Constraints{}, errors.Capture(err)
+	}
+
+	selectZoneStmt, err := st.Prepare(
+		"SELECT &dbConstraintZone.* FROM v_model_constraint_zone", dbConstraintZone{})
+	if err != nil {
+		return constraints.Constraints{}, errors.Capture(err)
+	}
+
+	var (
+		cons   dbConstraint
+		tags   []dbConstraintTag
+		spaces []dbConstraintSpace
+		zones  []dbConstraintZone
+	)
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		_, err := getModelUUID(ctx, st, tx)
+		if err != nil {
+			return errors.Errorf("checking if model exists: %w", err)
+		}
+
+		cons, err = st.getModelConstraints(ctx, tx)
+		if err != nil {
+			return errors.Capture(err)
+		}
+		err = tx.Query(ctx, selectTagStmt).GetAll(&tags)
+		if err != nil && !errors.Is(err, sqlair.ErrNoRows) {
+			return errors.Errorf("getting constraint tags: %w", err)
+		}
+		err = tx.Query(ctx, selectSpaceStmt).GetAll(&spaces)
+		if err != nil && !errors.Is(err, sqlair.ErrNoRows) {
+			return errors.Errorf("getting constraint spaces: %w", err)
+		}
+		err = tx.Query(ctx, selectZoneStmt).GetAll(&zones)
+		if err != nil && !errors.Is(err, sqlair.ErrNoRows) {
+			return errors.Errorf("getting constraint zones: %w", err)
+		}
+		return nil
+	})
+	if err != nil {
+		return constraints.Constraints{}, errors.Capture(err)
+	}
+
+	return cons.toValue(tags, spaces, zones)
+}
+
+func getModelUUID(ctx context.Context, preparer domain.Preparer, tx *sqlair.TX) (coremodel.UUID, error) {
```

Comment:
This should just be called `modelExists` and returns an error. No need to return the UUID if we're not going to use it.

Reply:
Yeah good call! Done.